lib/modules/manager/asdf/upgradeable-tooling.ts

```diff
@@ -0,0 +1,271 @@
+import { AdoptiumJavaDatasource } from '../../datasource/adoptium-java';
+import { DockerDatasource } from '../../datasource/docker';
+import { GithubReleasesDatasource } from '../../datasource/github-releases';
+import { GithubTagsDatasource } from '../../datasource/github-tags';
+import { NodeDatasource } from '../../datasource/node';
+import * as nodeVersioning from '../../versioning/node';
+import * as regexVersioning from '../../versioning/regex';
+import * as semverVersioning from '../../versioning/semver';
+import type { PackageDependency } from '../types';
+
+type StaticTooling = Pick<
+  PackageDependency,
+  'depName' | 'datasource' | 'packageName' | 'versioning' | 'extractVersion'
+>;
+
+type DynamicTooling = (version: string) => StaticTooling | undefined;
+
+export const upgradeableTooling: Record<
+  string,
+  StaticTooling | DynamicTooling
+> = {
+  awscli: {
+    datasource: GithubTagsDatasource.id,
+    packageName: 'aws/aws-cli',
+    versioning: semverVersioning.id,
+  },
+  bun: {
+    datasource: GithubReleasesDatasource.id,
+    packageName: 'oven-sh/bun',
+    versioning: semverVersioning.id,
+    extractVersion: '^bun-v(?<version>\\S+)',
+  },
+  'cargo-make': {
+    datasource: GithubReleasesDatasource.id,
+    packageName: 'sagiegurari/cargo-make',
+    versioning: semverVersioning.id,
+  },
+  clojure: {
+    datasource: GithubTagsDatasource.id,
+    packageName: 'clojure/brew-install',
+    versioning: `${regexVersioning.id}:^(?<major>\\d+?)\\.(?<minor>\\d+?)\\.(?<patch>\\d+)\\.(?<build>\\d+)$`,
+  },
+  crystal: {
+    datasource: GithubTagsDatasource.id,
```

Comment:
https://github.com/asdf-community/asdf-crystal/blob/f8347c072a53a55a89d25f59e2ab9e8990a566f7/bin/install#L39
```suggestion
    datasource: GithubReleasesDatasource.id,
```